README.md

```diff
@@ -82,78 +82,84 @@ deployment:
         mandatory: true
-        # enum=["red","yellow","green"]: the value must be in the list of enum. 
-        validator: enum=["red","yellow","green"]
-      - name: pattern_test
-        description: Validator pattern for value of dataType string.
-        kind: variable
-        type: string
-        mandatory: false
-        # pattern=*e*: the value matches any string that contains the specified. 
-        validator: pattern=*e*
+        validator: minLength=8|maxLength=16|pattern=^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{8,16}$
    deployer: |
-      variable "secgroup_id" {}
-      data "huaweicloud_availability_zones" "myaz" {}
-      data "huaweicloud_compute_flavors" "myflavor" {
-          availability_zone = data.huaweicloud_availability_zones.myaz.names[0]
-          performance_type  = "normal"
-          cpu_core_count    = 2
-          memory_size       = 4
+      data "huaweicloud_availability_zones" "osc-az" {}
+      variable "admin_passwd" {
+      type = string
+      description = "The root password for all nodes in the K8S cluster."
       }
-      data "huaweicloud_vpc_subnet" "mynet" {
-          name = "xpanse_subnet1"
+
+      variable "worker_nodes_count" {
+      type = number
+      description = "The number of worker nodes in the K8S cluster."
       }
-      resource "huaweicloud_compute_instance" "basic" {
-          name               = "basic"
-          image_id           = "a8601887-81d5-4eed-9338-382cf5b6d80b"
-          flavor_id          = data.huaweicloud_compute_flavors.myflavor.ids[0]
-          availability_zone  = data.huaweicloud_availability_zones.myaz.names[0]
-          network {
-              uuid = data.huaweicloud_vpc_subnet.mynet.id
-          }
+
+      variable "node_flavor_id" {
+      type = string
+      description = "The flavor id of all nodes in the K8S cluster."
+      }
+
+      resource "huaweicloud_compute_instance" "k8s-master" {
+      name = "k8s-master"
+      availability_zone = data.huaweicloud_availability_zones.osc-az.names[0]
+      flavor_id = var.node_flavor_id
+      network {
+          uuid = "6a7c406d-5bc6-4b50-8f75-948aea69cb24"
+      }
+      security_group_ids = [ "668833c8-459d-4d15-871d-34c3cbcda9d4" ]
+      image_id = "ef88c527-b556-4199-bae5-3e80fba1733e"
+      user_data=<<EOF
+      #!bin/bash
+      echo root:${var.admin_passwd} | sudo chpasswd
+      sh /root/k8s-init.sh true ${var.admin_passwd} ${var.worker_nodes_count} > /root/init.log
+      EOF
+      }
+
+      resource "huaweicloud_compute_instance" "k8s-node" {
+      count = var.worker_nodes_count
+      name = "k8s-node-${count.index}"
+      availability_zone = data.huaweicloud_availability_zones.osc-az.names[0]
+      flavor_id = var.node_flavor_id
+      network {
+          uuid = "6a7c406d-5bc6-4b50-8f75-948aea69cb24"
+      }
+
+      security_group_ids = [ "668833c8-459d-4d15-871d-34c3cbcda9d4" ]
+      image_id = "ef88c527-b556-4199-bae5-3e80fba1733e"
```

Comment:
avoid to use the `image_id` directly, we can use data resources to filter the right image with some special tags.

Reply:
Ok. I use data image to find image. Now this image can found in project 'osc' and 'default'